src/main/kotlin/com/criptext/mail/scenes/emaildetail/EmailDetailSceneController.kt

```diff
@@ -325,16 +325,14 @@ class EmailDetailSceneController(private val scene: EmailDetailScene,
     }
 
     private fun readEmails(emails: List<FullEmail>) {
-        val unreadEmails = emails.filter { it.email.unread }
-        if (unreadEmails.isNotEmpty()) {
-             val emailIds = unreadEmails.map { it.email.id }
-            val metadataKeys = unreadEmails.map { it.email.metadataKey }
+        val emailIds = emails.map { it.email.id }
+        val metadataKeys = emails.map { it.email.metadataKey }
+
+        dataSource.submitRequest(EmailDetailRequest.ReadEmails(
```

Comment:
So you always launch this task? Even if there are no new mails to read? Why? How does this solve an issue?

Reply:
There was an issue on the email detail view, the three dot menu, this menu checks the model to know if it needs to show 'mark as unread' or 'mark as read' when opening the menu. This was causing troubles since you could open the menu before returning from the ReadEmailsWorker and updating the model. So it was decided to change the LoadFullEmailsWorker to return a model with the unread field already modified, but, the actual update to the data base is handled by the ReadEmailsWorker. Even if it is called always, the worker internally checks if there is something that needs to be updated, if not it does nothing and just returns.